### ISSUE TITLE
Missing Test Case - 1203. Sort Items by Groups Respecting Dependencies

### ISSUE THREAD

**Issue by shashwat-nayak** [LeetCode-Feedback/LeetCode-Feedback#15753]
#### Your LeetCode username
shashwat_nayak 

#### Category of the bug
- [ ] Missing Test Cases 

#### Description of the bug
The Below written code passes all testcases and gives AC. But for certain testcases it will run into RUNTIME ERROR 

#### Code you used for Submit/Run operation

```
// 1203. Sort Items by Groups Respecting Dependencies
class Solution {
public:
    void solve(vector<int> &nodes,vector<int> &ans,int ind[],vector<int> a[]){
        queue<int> q;
        for(auto i:nodes)if(ind[i]==0)q.push(i);
        while(!q.empty()){
            auto node=q.front();
            q.pop();
            ans.push_back(node);
            for(auto i:a[node]){
                ind[i]--;
                if(ind[i]==0)q.push(i);
            }
        }
    }
    vector<int> sortItems(int n, int m, vector<int>& group, vector<vector<int>>& beforeItems) {
         vector<int> groups[30001],ans;
         vector<int> edges_node[30001],edges_grp[30001];
         int ind_grp[30001]={0},ind_node[30001]={0}; 
         for(int i=0;i<n;i++){
             if(group[i]==-1)group[i]=m++;
            groups[group[i]].push_back(i);
         }
         for(int i=0;i<n;i++){
             for(auto j:beforeItems[i]){
                 int g1=group[i];
                 int g2=group[j];
                 if(g1==g2){
                     ind_node[i]++;
                     edges_node[j].push_back(i);
                 }
                 else{
                     ind_grp[g1]++;
                     edges_grp[g2].push_back(g1);
                 }
             }
         }
         queue<int> q;
         for(int i=0;i<m;i++)if(ind_grp[i]==0)q.push(i);
         while(!q.empty()){
             auto grp=q.front();
             q.pop();
             solve(groups[grp],ans,ind_node,edges_node);
             for(auto i:edges_grp[grp]){
                 ind_grp[i]--;
                 if(ind_grp[i]==0)q.push(i);
             }
         }
        if(ans.size()!=n)ans.clear();
        return ans;
    }
};
```
[Submission LINK](https://leetcode.com/submissions/detail/1026704122/)

#### Language used for code
 C++ 

#### Expected behavior
 In the code, I have hardcoded the array size as 30001. In some testcases it can exceed the array size limit and give RUNTIME ERROR. According to my logic the good upper bound for array size is around 60001. But it gives AC verdict even with 30001. I have attached one such testcase here as character limit was getting exceeded when I tried to submit the testcase through leetcode.
Basically n=30000, m=30000, 
In group vector, 1st entry is 29999 and all other values are -1.
In beforeItems, there are 30000 number of vectors and all of them are empty. 
[testcase.txt](https://github.com/LeetCode-Feedback/LeetCode-Feedback/files/12388130/testcase.txt)

#### Screenshots

![1](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/75883118/d550b59d-dd1a-4d62-a537-925b6c92f314)
I have added the testcase for sample run.

![2](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/75883118/0a42bea4-ff06-4ff5-8595-4d13bf689d9d)
It gives Runtime error on sample run.

![3](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/75883118/25f82578-9afe-4043-8330-81480512b359)
Still one can get AC using the same code.

**Comment by LC-Pam** [comment]
Hi there,

Thank you for reaching out to us.

 We've relayed this issue to our team to investigate. 

We will reach back to you after we have the results.

Thank you for your support.

**Comment by LC-Pam** [comment]
Thank you for your time. 

We've used your feedback to update the problem. 

Your LeetCode account has received 100 LeetCoins as a reward for this feedback. 

If you have any other questions or feedback, please don't hesitate to let us know! 

We appreciate your support!
